### ISSUE TITLE
Make pkg/util/k8sutil functions private or internal

### ISSUE THREAD

**Issue by ericchiang** [operator-framework/operator-sdk#160]
If this isn't intended to be used by consumers we probably shouldn't allow them to use it. It's just extra API that needs to be maintained.

**Comment by ericchiang** [comment]
cc @Verolop

**Comment by hasbro17** [comment]
Blocked by #169

**Comment by pawelprazak** [comment]
IMHO most important functions in `pkg/k8sclient` should be "configurable", as in an interface that can have a custom implementation, something like a "kubeclient factory", that would allow an SDK user to implement features like #351

**Comment by hasbro17** [comment]
After #382 we've removed the k8sutil and k8sclient helper functions that weren't supposed to be exposed as an API to users.

@pawelprazak I've addressed how the custom client functionality needed by #351 can be added on the PR.  

`pkg/generator` is now `pkg/scaffold` whose utils are mostly internal now.
But there is something to be said for making `pkg/scaffold` completely internal since that's not something the SDK users need to see as a potential API(not that anyone has done so far). It's only needed by the SDK CLI in `cmd/...`.
@estroz WDYT? We can follow that up in a separate issue though since this one was mostly related to the unnecessary client helpers we exposed initially.
